File: backend/app/models/professional.py
import uuid

from sqlalchemy import Boolean, ForeignKey, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin, UUIDPKMixin


class Professional(UUIDPKMixin, TimestampMixin, Base):
    __tablename__ = "professionals"

    tenant_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False
    )
    name: Mapped[str] = mapped_column(String, nullable=False)
    phone: Mapped[str | None] = mapped_column(String, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)

File: app_admin/lib/models/professional.dart
class Professional {
  const Professional({
    required this.id,
    required this.name,
    required this.isActive,
    this.phone,
  });

  final String id;
  final String name;
  final bool isActive;
  final String? phone;

  factory Professional.fromJson(Map<String, dynamic> json) {
    return Professional(
      id: json['id'] as String,
      name: json['name'] as String,
      isActive: json['is_active'] as bool,
      phone: json['phone'] as String?,
    );
  }
}
